crates/bevy_window/src/window.rs

```diff
@@ -479,9 +500,17 @@ impl WindowPosition {
 )]
 #[reflect(Debug, PartialEq, Default)]
 pub struct WindowResolution {
+    /// Width of the window in physical pixels.
     physical_width: u32,
+    /// Height of the window in physical pixels.
     physical_height: u32,
+    /// Ratio for `physical_with` (or `physical_height`) over (requested) `width` (or `height`).
+    ///
+    /// Will override `scale_factor`.
     scale_factor_override: Option<f64>,
+    /// Ratio for `physical_with` (or `physical_height`) over logical width (or height).
+    ///
+    /// Set by the operating system depending on monitor pixel densities.
```

Comment:
I don't understand that sentence.

Reply:
I should just rewrite it "Ratio for physical size to requested size" (respectively logical)

physical/requested/logical is explained above on the doc of WindowResolution

I didn't modify that explanation, mainly because I'm not sure I understand it, but I feel like it could be made more clear!

Reply:
Partially resolved by https://github.com/bevyengine/bevy/pull/8858/commits/a36128c631ab0677c3cfd13e09ff2026af4b5315 (?) but WindowResolution's doc should probably be rewritten.

Reply:
Oh I think I understand now, nevermind.

Maybe something not involving the OS? It's kind of a detail; the user only cares that's it's automatically set, no?

```suggestion
    /// Set automatically depending on the pixel density of the screen.
```

Reply:
Commited your change but keeping the conversation open to clarify `WindowResolution`'s doc

Reply:
Thanks

Reply:
After reading more of the code and pocking around with the `scale_factor_override` example, here's my attempt to clarify `WindowResolution`'s documentation: https://github.com/bevyengine/bevy/pull/8858/commits/5a303192852b62ab72d886d898ca870cb3d41b8e

I tried to stay somewhat general and focus more on the intent than the actual behaviour, because I feel like some behaviours are not necessarily intended, for example in the scale_factor_override example if the scale factor is too big to fit on the screen, the logical size will change to match the ratio of height to width* but the physical size will continue to increase despite the window not taking all of that space, I feel like this is not intended for example.

*(which is what I'm talking about in `The requested size is not kept in memory, for example changing the scale factor one way and changing it back after might not get the original size back.`, I didn't stay general in that part)

Edit: (changed that last part to: `The requested size is not kept in memory, for example requesting a size too big for the screen, making the logical size different from the requested size, then setting a scale factor that makes the previous requested size within the limits of the screen will not get back that previous requested size.`)